### DISCUSSION TITLE
How to query all transaction information?

### DISCUSSION THREAD

**Discussion by WuTao18** [near/near-indexer-for-explorer#175] [Q&A] [answered]
I deployed NEAR Indexer for Explorer on my server, and I want to query any transaction data.

First, I used JSON RPC to query, and I could only query transaction information submitted within 5 epochs, because I deployed  NEAR Indexer for Explorer as non-archival mode.

Then I tried to query the PostgreSQL database, but only the data after sync started can be queried. I use the snapshot to speed up the synchronization, but only missing data was added to the PostgreSQL database. What data was included in the snapshot?

I also tried to synchronize without the snapshot (the synchronization was not completed). Why is there no data in the PostgreSQL database? Is it related to the warning in the figure below?

![image](https://user-images.githubusercontent.com/32795326/136684458-6cf8a1ae-c2d7-4a11-afda-84db002ca16d.png)

Is it necessary to deploy to archival mode to query any transaction information?

**Comment by telezhnaya** [comment, marked as answer]
You can ignore the warning you are highlighting. It's only about `aggregated__circulating_supply` table.

If you want to query "any transaction data" for the all time period, I'm afraid you need to run the Indexer in archival mode from the early beginning. For Mainnet, be ready to spend ~1TB of storage and at least one week of pure script work to collect all the data.

**Reply by WuTao18** [reply]
I see, thank u!

**Comment by frol** [comment]
>  Why is there no data in the PostgreSQL database?

The node should sync block headers first to have at least some data to index. Syncing block headers and blocks themselves from genesis will most likely take months at this point, so we recommend using snapshots. (chain team is working on sync improvements, but they are not there yet)

**Reply by WuTao18** [reply]
Thanks for your reply!
